mmlearn/modules/layers/attention.py

```diff
@@ -0,0 +1,266 @@
+    """
+    Drop paths (Stochastic Depth) for regularization during training.
+
+    Parameters
+    ----------
+    x : torch.Tensor
+        Input tensor.
+    drop_prob : float, optional, default=0.0
+        Probability of dropping paths.
+    training : bool, optional, default=False
+        Whether the model is in training mode.
+
+    Returns
+    -------
+    output : torch.Tensor
+        Output tensor after applying drop path.
+    """
+    if drop_prob == 0.0 or not training:
+        return x
+    keep_prob = 1 - drop_prob
+    shape = (x.shape[0],) + (1,) * (x.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = keep_prob + torch.rand(shape, dtype=x.dtype, device=x.device)
+    random_tensor.floor_()  # binarize
+    output = x.div(keep_prob) * random_tensor
+    return output
+
+
+class DropPath(nn.Module):
+    """
+    Drop paths (Stochastic Depth) per sample (when applied in the main path of residual blocks).
+
+    Parameters
+    ----------
+    drop_prob : Optional[float], optional
+        Probability of dropping paths. Default is None.
+    """
+
+    def __init__(self, drop_prob: Optional[float] = None) -> None:
+        super(DropPath, self).__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward pass through DropPath module."""
+        return drop_path(x, self.drop_prob, self.training)
+
+
+class MLP(nn.Module):
+    """
+    Multi-layer Perceptron (MLP).
+
+    A simple feedforward neural network with one hidden layer and an activation function.
+
+    Parameters
+    ----------
+    in_features : int
+        Number of input features.
+    hidden_features : Optional[int], optional
+        Number of hidden features. If None, defaults to in_features. Default is None.
+    out_features : Optional[int], optional
+        Number of output features. If None, defaults to in_features. Default is None.
+    act_layer : Callable[..., nn.Module], optional
+        Activation function to use. Default is nn.GELU.
+    drop : float, optional
+        Dropout probability. Default is 0.0.
+    """
+
+    def __init__(
+        self,
+        in_features: int,
+        hidden_features: Optional[int] = None,
+        out_features: Optional[int] = None,
+        act_layer: Callable[..., nn.Module] = nn.GELU,
+        drop: float = 0.0
+    ) -> None:
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.fc1 = nn.Linear(in_features, hidden_features)
+        self.act = act_layer()
+        self.fc2 = nn.Linear(hidden_features, out_features)
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward pass through the MLP."""
+        x = self.fc1(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
```

Comment:
Let's use the [existing MLP implementation in `mmlearn`](https://github.com/VectorInstitute/mmlearn/blob/main/mmlearn/modules/layers/mlp.py) instead of creating a new one. I understand that it'll mean we won't be able to load ijepa weights directly, but we can try to find a way around that later.

Reply:
Done!